test/iht/views/application/ApplicationPageBehaviour.scala

```diff
@@ -84,7 +86,11 @@ trait ApplicationPageBehaviour extends ViewTestHelper {
       "show the return link with the correct target and text" in {
         cancelComponent.foreach { attrib =>
           val cancelButton = doc.getElementById(cancelId)
-          cancelButton.attr("href") shouldBe attrib.target.url
+          if(attrib.hash > ""){
```

Comment:
Better, clearer, to use .nonEmpty here. Personally I'd probably write it like this:
          val expectedResult = if(attrib.hash.nonEmpty) attrib.target.url + attrib.hash else attrib.target.url
          cancelButton.attr("href") shouldBe expectedResult
but that's perhaps more a matter of personal taste.
Also, this bit of code is repeated 4 times throughout tests, so I'd put it somewhere common such as TestHelper.